### ISSUE TITLE
Hono streams immediately closes the connection unless sleep is used

### ISSUE THREAD

**Issue by satyavh** [honojs/hono#2050]
### What version of Hono are you using?

3.12.0

### What runtime/platform is your app running on?

Bun

### What steps can reproduce the bug?

According to the Hono documentation, to setup a stream and send SSE events, this is all that's needed

```
const app = new Hono()
let id = 0

app.get('/sse', async (c) => {
  return streamSSE(c, async (stream) => {
    while (true) {
      const message = `It is ${new Date().toISOString()}`
      await stream.writeSSE({
        data: message,
        event: 'time-update',
        id: String(id++),
      })
      await stream.sleep(1000)
    }
  })
})
```

While that is correct, the reason this works is because of the magic line `await stream.sleep(1000)` which basically keeps the connection open forever, extending it with 1000ms every loop. Basically the streamSSE callback never resolves.

Without this, Hono immediately resolves the callback, returns and closes the connection. And no more SSE events will be pushed.

In other words Hono does not keep the connection open unless you manually keep it open using some `await stream.sleep()` mechanism.

And because a Hono request always requires a return, there's no other option then to increase the time with some sleep.

### What is the expected behavior?

I expect Hono to be able to keep a stream connection open so I can push future updates to it, without using stream.sleep.

### What do you see instead?

Connection getting closed if I don't use await.stream.sleep.

**Comment by yusukebe** [comment]
@satyavh Thanks for the issue.

@sor4chi Can you handle this?

**Comment by sor4chi** [comment]
Hi, @satyavh 

Is this the code you expect to work correctly?

```
import { Hono } from "hono";
import { streamSSE } from "hono/streaming";

const app = new Hono();
let id = 0;

app.get("/sse", async (c) => {
  return streamSSE(c, async (stream) => {
    while (true) {
      const message = `It is ${new Date().toISOString()}`;
      await stream.writeSSE({
        data: message,
        event: "time-update",
        id: String(id++),
      });
      // await stream.sleep(1000);
    }
  });
});

export default app;
```

**Comment by satyavh** [comment]
Exactly

And then you'd always manually need to close the stream using `stream.close()` - or of course from the client side

**Comment by watany-dev** [comment]
If that is the problem closing to streamSSE, the v3.12.1 update may have improved it.

**Comment by abhi12299** [comment]
This can be closed - works fine now. Also the bug label can be removed too.

**Comment by yusukebe** [comment]
Okay. Closing. Thanks!
